src/HttpClientRequestAdapter.cs

```diff
@@ -293,7 +295,70 @@ public ISerializationWriterFactory SerializationWriterFactory
                         {
                             result = rootNode.GetDateValue();
                         }
-                        else throw new InvalidOperationException("error handling the response, unexpected type");
+                        else if(
+                            Nullable.GetUnderlyingType(modelType) is { IsEnum: true } underlyingType &&

```

Comment:
actually, we already have all the logic implemented here (and in other format providers)
Couldn't we simply call rootNode.GetEnumValue?

https://github.com/microsoft/kiota-serialization-json-dotnet/blob/d0454daecb7ec2327f3e55d642fa510cf617e8b5/src/JsonParseNode.cs#L214

Reply:
Due to the type constraint in the method definition for `getEnumValue` as `where T : struct, Enum;`, we unfortunately can't do that as the generic type parameter for this function won't match the constraint.

Reply:
right, now I remember getting to this limitation in the early days of kiota, thinking we might need another request adapter method for enums, and forgetting about it 🤦‍♂️
Resolving